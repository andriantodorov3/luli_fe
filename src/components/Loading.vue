<template>
  <div class="loading pa-10">
    <v-progress-circular indeterminate color="purple" class="mr-3"></v-progress-circular>
    <slot>Loading...</slot>
  </div>
</template>

<script>
export default {
  name: "NoResults",
};
</script>

<style>
.loading {
  text-align: center;
}
</style>