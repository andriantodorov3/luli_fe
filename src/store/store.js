import Vue from 'vue'
import Vuex from 'vuex'

import { state } from '@/store/vuex.state'
import { getters } from '@/store/vuex.getters'
import { mutations } from '@/store/vuex.mutations'
import { actions } from '@/store/vuex.actions'

Vue.use(Vuex)

export default new Vuex.Store({
    state,
    getters,
    mutations,
    actions
})