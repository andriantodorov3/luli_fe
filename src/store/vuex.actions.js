export const actions = {
    notification({ state }, payload) {
        state.bus.$emit('shownotification', payload);

    },

}