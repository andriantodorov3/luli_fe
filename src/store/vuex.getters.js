import { getField } from "vuex-map-fields";
export const getters = {
    getField,
    bus: state => state.bus,
}