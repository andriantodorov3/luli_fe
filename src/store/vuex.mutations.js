import { updateField } from "vuex-map-fields";

export const
    mutations = {
        updateField,
    };