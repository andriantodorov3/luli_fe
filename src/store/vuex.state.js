import Vue from 'vue'
export const state = {
    //event bus
    bus: new Vue(),
    //left navi 
    drawer: null,
    mini: false,

}