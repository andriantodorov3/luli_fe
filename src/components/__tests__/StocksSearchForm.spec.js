import { shallowMount, mount, createLocalVue } from '@vue/test-utils'
import Komponent from '@/components/StockSearchForm.vue'
import Loading from '@/components/Loading.vue';

import store from '@/store/store.js'
import Vuetify from 'vuetify'

import Vue from 'vue';

import flushPromises from 'flush-promises';

import VueRouter from 'vue-router'

describe('Loading checks', () => {
    it('Submit btn is disabled when loading', () => {
        const props = {
            end_errors: [],
            start_errors: [],
            loading: true,
        };

        const wrapper = shallowMount(Komponent, {
            propsData: props
        })
        expect(wrapper.find('#search-btn').attributes().disabled).toBeDefined()
    })

    it('Submit btn is NOT disabled with default form content', () => {
        const props = {
            end_errors: [],
            start_errors: [],
            loading: false,
        };

        const wrapper = shallowMount(Komponent, {
            propsData: props
        })
        expect(wrapper.find('#search-btn').attributes().disabled).not.toBeDefined()
    })


    it('Loading component exists when loading true', () => {
        const props = {
            end_errors: [],
            start_errors: [],
            loading: true,
        };

        const wrapper = shallowMount(Komponent, {
            propsData: props
        })
        expect(wrapper.findComponent(Loading).exists()).toBe(true)
    })

    it('Loading component does not exist when loading false', () => {
        const props = {
            end_errors: [],
            start_errors: [],
            loading: false,
        };

        const wrapper = shallowMount(Komponent, {
            propsData: props
        })
        expect(wrapper.findComponent(Loading).exists()).toBe(false)
    })

    it('Button has text search when not loading', () => {
        const props = {
            end_errors: [],
            start_errors: [],
            loading: false,
        };

        const wrapper = shallowMount(Komponent, {
            propsData: props
        })
        expect(wrapper.find('#search-btn').text()).toBe('Search')
    })

})

describe('Form validations', () => {

    const localVue = createLocalVue()
    let vuetify

    const router = new VueRouter()

    localVue.use(VueRouter)

    beforeEach(() => {
        vuetify = new Vuetify()
    })


    const validStartDate = '2022-08-10T20:48';
    const validEndDate = '2022-08-12T20:48';
    const validFunds = 100;

    it('Check negative funds input', async() => {
        const props = {
            end_errors: [],
            start_errors: [],
            loading: false,
        };

        const wrapper = mount(Komponent, {
            propsData: props,
            localVue,
            vuetify,
            store,
            router,

        })
        wrapper.find('#funds_input').setValue(-1);
        wrapper.find('#date_picker_start').setValue(validStartDate);
        wrapper.find('#date_picker_end').setValue(validEndDate);

        await Vue.nextTick();
        expect(wrapper.vm.formIsValid).toBeFalsy();
    })


    it('Check 0 funds input', async() => {
        const props = {
            end_errors: [],
            start_errors: [],
            loading: false,
        };

        const wrapper = mount(Komponent, {
            propsData: props,
            localVue,
            vuetify,
            store,
            router,

        })
        wrapper.find('#funds_input').setValue(0);
        wrapper.find('#date_picker_start').setValue(validStartDate);
        wrapper.find('#date_picker_end').setValue(validEndDate);

        await Vue.nextTick();
        expect(wrapper.vm.formIsValid).toBeFalsy();
    })

    it('Check 100 funds input', async() => {
        const props = {
            end_errors: [],
            start_errors: [],
            loading: false,
        };

        const wrapper = mount(Komponent, {
            propsData: props,
            localVue,
            vuetify,
            store,
            router,

        })
        wrapper.find('#funds_input').setValue(validFunds);
        wrapper.find('#date_picker_start').setValue(validStartDate);
        wrapper.find('#date_picker_end').setValue(validEndDate);

        await Vue.nextTick();
        expect(wrapper.vm.formIsValid).toBeTruthy();
    })



    it('Check invalid date order', async() => {
        const props = {
            end_errors: [],
            start_errors: [],
            loading: false,
        };

        const wrapper = mount(Komponent, {
            propsData: props,
            localVue,
            vuetify,
            store,
            router,

        })
        wrapper.find('#funds_input').setValue(validFunds);
        wrapper.find('#date_picker_end').setValue(validStartDate);
        wrapper.find('#date_picker_start').setValue(validEndDate);

        await Vue.nextTick();
        expect(wrapper.vm.formIsValid).toBeFalsy();
    })

})

describe('Form events', () => {

    const localVue = createLocalVue()
    let vuetify

    const router = new VueRouter()

    localVue.use(VueRouter)

    beforeEach(() => {
        vuetify = new Vuetify()
    })


    const validStartDate = '2022-08-10T20:48';
    const validEndDate = '2022-08-12T20:48';
    const validFunds = 100;

    it('Check form emits search for valid input', async() => {
        const props = {
            end_errors: [],
            start_errors: [],
            loading: false,
        };

        const wrapper = mount(Komponent, {
            propsData: props,
            localVue,
            vuetify,
            store,
            router,

        })
        wrapper.find('#funds_input').setValue(validFunds);
        wrapper.find('#date_picker_start').setValue(validStartDate);
        wrapper.find('#date_picker_end').setValue(validEndDate);

        await Vue.nextTick();
        expect(wrapper.vm.formIsValid).toBeTruthy();


        await wrapper.find('#search-btn').trigger('click');

        expect(wrapper.emitted()).toHaveProperty('search')
    })




})

describe('Snapshot', () => {

    const localVue = createLocalVue()
    let vuetify

    const router = new VueRouter()

    localVue.use(VueRouter)

    beforeEach(() => {
        vuetify = new Vuetify()
    })


    it('Checks snapshot', async() => {

        const props = {
            end_errors: [],
            start_errors: [],
            loading: false,
        };

        const wrapper = mount(Komponent, {
            propsData: props,
            localVue,
            vuetify,
            store,
            router,

        })

        await flushPromises();

        expect(wrapper.element).toMatchSnapshot();
    })
});