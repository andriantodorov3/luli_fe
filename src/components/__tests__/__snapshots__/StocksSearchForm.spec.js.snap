// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Checks snapshot 1`] = `
<form
  class="v-form"
  novalidate="novalidate"
>
  <div
    class="container new-capsule-wrapper lighten-5"
  >
    <div
      class="row space-around align-center justify-space-around"
    >
      <div
        class="pt-2 col-sm-6 col-md-4 col-lg-3 col"
      >
        <i
          aria-hidden="true"
          class="v-icon notranslate mr-2 mdi mdi-clock-time-four-outline theme--light"
        />
         
        <label>
           Select timeframe 
        </label>
      </div>
       
      <div
        class="col-sm-6 col-md-8 col-lg-9 col"
      >
        <div
          class="v-input d-inline-flex ml-2 v-input--is-label-active v-input--is-dirty v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined"
        >
          <div
            class="v-input__control"
          >
            <div
              class="v-input__slot"
            >
              <fieldset
                aria-hidden="true"
              >
                <legend
                  style="width: 0px;"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
              <div
                class="v-text-field__slot"
              >
                <label
                  class="v-label v-label--active theme--light"
                  for="date_picker_start"
                  style="left: 0px; position: absolute;"
                >
                  Select start date
                </label>
                <input
                  id="date_picker_start"
                  type="datetime-local"
                />
              </div>
            </div>
            <div
              class="v-text-field__details"
            >
              <div
                class="v-messages theme--light"
              >
                <transition-group-stub
                  class="v-messages__wrapper"
                  name="message-transition"
                  tag="div"
                />
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="v-input d-inline-flex ml-2 v-input--is-label-active v-input--is-dirty v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined"
        >
          <div
            class="v-input__control"
          >
            <div
              class="v-input__slot"
            >
              <fieldset
                aria-hidden="true"
              >
                <legend
                  style="width: 0px;"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
              <div
                class="v-text-field__slot"
              >
                <label
                  class="v-label v-label--active theme--light"
                  for="date_picker_end"
                  style="left: 0px; position: absolute;"
                >
                  Select end date
                </label>
                <input
                  id="date_picker_end"
                  type="datetime-local"
                />
              </div>
            </div>
            <div
              class="v-text-field__details"
            >
              <div
                class="v-messages theme--light"
              >
                <transition-group-stub
                  class="v-messages__wrapper"
                  name="message-transition"
                  tag="div"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="row align-start justify-start"
    >
      <div
        class="col-sm-12 col-md-12 col-lg-12 col"
      >
        <div
          class="v-input v-input--is-label-active v-input--is-dirty v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined v-text-field--placeholder"
        >
          <div
            class="v-input__control"
          >
            <div
              class="v-input__slot"
            >
              <fieldset
                aria-hidden="true"
              >
                <legend
                  style="width: 0px;"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
              <div
                class="v-text-field__slot"
              >
                <label
                  class="v-label v-label--active theme--light"
                  for="funds_input"
                  style="left: 0px; position: absolute;"
                >
                  Enter funds
                </label>
                <input
                  id="funds_input"
                  min="1"
                  step="1"
                  type="number"
                />
                <div
                  class="v-text-field__suffix"
                >
                  USD
                </div>
              </div>
            </div>
            <div
              class="v-text-field__details"
            >
              <div
                class="v-messages theme--light"
              >
                <transition-group-stub
                  class="v-messages__wrapper"
                  name="message-transition"
                  tag="div"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="row align-start justify-start"
    >
      <div
        class="col-sm-12 col-md-12 col-lg-12 col"
      >
        <button
          class="submit-btn v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--x-large blue"
          id="search-btn"
          type="button"
        >
          <span
            class="v-btn__content"
          >
             Search 
          </span>
        </button>
      </div>
    </div>
  </div>
</form>
`;
